python/cuspatial/cuspatial/utils/column_utils.py

```diff
@@ -97,3 +97,21 @@ def contains_only_polygons(gs: GeoSeries):
     """
 
     return contain_single_type_geometry(gs) and len(gs.polygons.xy) > 0
+
+
+def has_same_geometry(lhs: GeoSeries, rhs: GeoSeries):
+    """
+    Returns true if `lhs` and `rhs` have only features of the same homogeneous
+    geometry type.
+    """
+
+    if contains_only_points(lhs) and contains_only_points(rhs):
+        return True
+    elif contains_only_multipoints(lhs) and contains_only_multipoints(rhs):
+        return True
+    elif contains_only_linestrings(lhs) and contains_only_linestrings(rhs):
+        return True
+    elif contains_only_polygons(lhs) and contains_only_polygons(rhs):
+        return True
+    else:
+        return False
```

Comment:
Seems this would be more efficient and simpler if you first confirm that each side has a single geometry type, and then compare the geometry type. As it is, `contains_only_*` re-confirms that there is only a single geometry type each time.  If you can also provide a way to just get a value representing the geometry types contained in a GS, then you only have to check that one of them has only a single geometry type, and then compare the types codes of both sides.

```suggestion
    if contains_single_geometry_type(lhs):
      return geometry_types(lhs) == geometry_types(rhs)
    else:
      return False
```

Reply:
I don't think we have a `geometry_types` function yet, what does it do?

Reply:
It's hypothetical. The idea is to return something that enables uniquely identifying all the geometry types contained in the series, so they can be compared. E.g. a bitmask.

Reply:
We can probably depend on pandas ExtensionDtype and make a `GeometryDtype` and assign to `dtype`. We can assign more meta data to `GeometryDtype`, for example, the underlying mixed geometry type. We can model `GeometryDtype` to `UnionType`.

Reply:
Should this be a separate effort? We can create an issue from this thread if so.